app/uk/gov/hmrc/thirdpartyapplication/repository/ApplicationRepository.scala

```diff
@@ -128,6 +128,9 @@ class ApplicationRepository @Inject()(mongo: ReactiveMongoComponent)(implicit va
   def updateApplicationIpWhitelist(applicationId: UUID, ipWhitelist: Set[String]): Future[ApplicationData] =
     updateApplication(applicationId, Json.obj("$set" -> Json.obj("ipWhitelist" -> ipWhitelist)))
 
+  def updateClientSecrets(applicationId: UUID, clientSecrets: Seq[ClientSecret]): Future[ApplicationData] =
+    updateApplication(applicationId, Json.obj("$set" -> Json.obj("tokens.production.clientSecrets" -> clientSecrets)))
```

Comment:
Not sure about this - going to overwrite the entire `clientSecrets` collection...

Reply:
You should use the positional operator I think: https://docs.mongodb.com/manual/reference/operator/update/positional/

Reply:
It's a once off, probably not a big deal. But I'm going to see if I can make it work with that positional operator.